extensions/markdown-language-features/src/languageFeatures/copyFiles/shared.ts

```diff
@@ -56,13 +56,52 @@ export const mediaMimes = new Set([
 	'audio/x-wav',
 ]);
 
+export async function getMarkdownLink(document: vscode.TextDocument, ranges: readonly vscode.Range[], dataTransfer: vscode.DataTransfer, token: vscode.CancellationToken): Promise<{ additionalEdits: vscode.WorkspaceEdit; label: string } | undefined> {
+	if (ranges.length === 0) {
+		return;
+	}
+
+	const edits: vscode.SnippetTextEdit[] = [];
+	let placeHolderValue: number = ranges.length;
+	let label: string = '';
+
+	for (let i = 0; i < ranges.length; i++) {
+		const snippet = await tryGetUriListSnippet(document, dataTransfer, token, document.getText(ranges[i]), placeHolderValue) || await tryGetUriListLinkSnippet(document, dataTransfer, token, document.getText(ranges[i]), placeHolderValue);
```

Comment:
This changes the behavior of the existing file pasting logic, which we want to avoid

Maybe instead of passing the entire data transfer to this function, `getMarkdownLink` can take a string value with the uris?